jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Retainable.java

```diff
@@ -26,6 +26,19 @@
  */
 public interface Retainable
 {
+    /**
+     * <p>Returns whether this resource can be retained, that is whether {@link #retain()}
+     * can be called safely.</p>
+     * <p>Implementations may decide that special resources are not retainable (for example,
+     * {@code static} constants) so calling {@link #retain()} is not safe because it may throw.</p>
+     *
+     * @return whether it is safe to call {@link #retain()}
+     */
+    default boolean canRetain()
```

Comment:
I would not provide a default impl, this is worth explicitly stating in all implementations.